include/libhal-input/ch9329.hpp

```diff
@@ -26,12 +28,70 @@ namespace hal::input {
  */
 class ch9329
 {
+
+public:
+  struct chip_info
+  {
+    hal::byte version;
+    bool enumeration_status;
+    bool num_lock;
+    bool caps_lock;
+    bool scroll_lock;
+  };
+
+  struct usb_string_descriptor
+  {
+    auto full_span()
+    {
+      return std::span(buffer);
+    }
+    auto received_data()
+    {
+      return std::span(buffer).first(length);
+    }
+    std::array<hal::byte, 23> buffer{};
```

Comment:
Create a `constexpr` value for each magic number we have in the code. Makes it easier to understand why something is length 23.